src/test/java/htsjdk/variant/bcf2/BCF2WriterUnitTest.java

```diff
@@ -0,0 +1,253 @@
+        additionalColumns.add("extra1");
+        additionalColumns.add("extra2");
+        final VCFHeader ret = new VCFHeader(metaData, additionalColumns);
+        ret.addMetaDataLine(new VCFInfoHeaderLine("GT", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFInfoHeaderLine("DP", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFInfoHeaderLine("BB", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFInfoHeaderLine("GQ", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFFormatHeaderLine("GT", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFFormatHeaderLine("DP", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFFormatHeaderLine("BB", 1, VCFHeaderLineType.String, "x"));
+        ret.addMetaDataLine(new VCFFormatHeaderLine("GQ", 1, VCFHeaderLineType.String, "x"));
+        ret.setSequenceDictionary(sequenceDict);
+        return ret;
+    }
+
+    @BeforeClass
+    private void createTemporaryDirectory() {
+        tempDir = TestUtil.getTempDirectory("BCFWriter", "StaleIndex");
+    }
+
+    @AfterClass
+    private void deleteTemporaryDirectory() {
+        for (File f : tempDir.listFiles()) {
+            f.delete();
+        }
+        tempDir.delete();
+    }
+
+    /**
+     * test, using the writer and reader, that we can output and input BCF without problems
+     */
+    @Test
+    public void testWriteAndReadBCF() throws IOException {
+        final File fakeVCFFile = VariantBaseTest.createTempFile("testWriteAndReadVCFBody.", ".bcf");
+
+        Tribble.indexFile(fakeVCFFile).deleteOnExit();
+        metaData = new HashSet<VCFHeaderLine>();
+        additionalColumns = new HashSet<String>();
+        final SAMSequenceDictionary sequenceDict = createArtificialSequenceDictionary();
+        final VCFHeader header = createFakeHeader(metaData, additionalColumns, sequenceDict);
+        try (final VariantContextWriter writer = new VariantContextWriterBuilder()
+                .setOutputFile(fakeVCFFile).setReferenceDictionary(sequenceDict)
+                .setOptions(EnumSet.of(Options.ALLOW_MISSING_FIELDS_IN_HEADER, Options.INDEX_ON_THE_FLY))
+                .build()) {
+            writer.writeHeader(header);
+            writer.add(createVC(header));
+            writer.add(createVC(header));
+        }
+        VariantContextTestProvider.VariantContextContainer container = VariantContextTestProvider
+                .readAllVCs(fakeVCFFile, new BCF2Codec());
+        int counter = 0;
+        final Iterator<VariantContext> it = container.getVCs().iterator();
+        while (it.hasNext()) {
+            it.next();
+            counter++;
+        }
+        Assert.assertEquals(counter, 2);
+    }
+
+    /**
+     * test, using the writer and reader, that we can output and input a BCF body without header
+     */
+    @Test
+    public void testWriteAndReadBCFBody() throws IOException {
```

Comment:
I just noticed a previous reviewer suggest the name "Body". I like Headerless a bit better since I don't think "body" is used this way in htsjdk.